src/transformers/models/swinv2/modeling_swinv2.py

```diff
@@ -898,12 +898,22 @@ def custom_forward(*inputs):
                 layer_outputs = layer_module(hidden_states, input_dimensions, layer_head_mask, output_attentions)
 
             hidden_states = layer_outputs[0]
-            output_dimensions = layer_outputs[1]
+            hidden_states_before_downsampling = layer_outputs[1]
+            output_dimensions = layer_outputs[2]
 
             input_dimensions = (output_dimensions[-2], output_dimensions[-1])
-            all_input_dimensions += (input_dimensions,)
 
-            if output_hidden_states:
+            if output_hidden_states and self.config.output_hidden_states_before_downsampling:
```

Comment:
Could you simplify the if-else statement over here? Maybe we could move the content within another method.
```suggestion
            output_before_downsampling = self.config.output_hidden_states_before_downsampling
            if output_hidden_states and output_before_downsampling:
                 all_reshaped_hidden_states = self.reshape_features(hidden_states_before_downsampling)
            elif ... :
```